### ISSUE TITLE
0.12.0: pytest is failing in one unit

### ISSUE THREAD

**Issue by kloczek** [beartype/beartype#213]
I'm packaging your module as an rpm package so I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with $PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>
- build is performed in env which is *`cut off from access to the public network`* (pytest is executed with `-m "not network"`)

Probably failt is caused by use lates sphinx.
Here is pytest output:
<details>

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-beartype-0.12.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-beartype-0.12.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra -m 'not network' -q
============================= test session starts ==============================
platform linux -- Python 3.8.16, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/beartype-0.12.0, configfile: pytest.ini, testpaths: beartype_test
collected 263 items

beartype_test/a00_unit/a00_core/test_a00_package.py ..                   [  0%]
beartype_test/a00_unit/a00_core/test_a90_typing.py .                     [  1%]
beartype_test/a00_unit/a10_data/hint/pep/sign/test_datapepsignset.py .   [  1%]
beartype_test/a00_unit/a20_util/test_utilobject.py ..                    [  2%]
beartype_test/a00_unit/a20_util/a00_cache/test_utilcachecall.py ...      [  3%]
beartype_test/a00_unit/a20_util/a00_cache/map/test_utilmapbig.py .       [  3%]
beartype_test/a00_unit/a20_util/a00_cache/map/test_utilmaplru.py ...     [  4%]
beartype_test/a00_unit/a20_util/a00_cache/pool/test_utilcachepool.py ..  [  5%]
beartype_test/a00_unit/a20_util/a00_cache/pool/test_utilcachepoollistfixed.py . [  6%]
...                                                                      [  7%]
beartype_test/a00_unit/a20_util/a00_cache/pool/test_utilcachepoolobjecttyped.py . [  7%]
.                                                                        [  7%]
beartype_test/a00_unit/a20_util/cls/test_utilclstest.py ...              [  9%]
beartype_test/a00_unit/a20_util/cls/pep/test_utilpep3119.py ..           [  9%]
beartype_test/a00_unit/a20_util/cls/pep/test_utilpep557.py .             [ 10%]
beartype_test/a00_unit/a20_util/error/test_utilerror.py .                [ 10%]
beartype_test/a00_unit/a20_util/func/test_utilfunccode.py .              [ 11%]
beartype_test/a00_unit/a20_util/func/test_utilfunccodeobj.py ..          [ 11%]
beartype_test/a00_unit/a20_util/func/test_utilfuncfile.py ..             [ 12%]
beartype_test/a00_unit/a20_util/func/test_utilfuncframe.py ..            [ 13%]
beartype_test/a00_unit/a20_util/func/test_utilfuncget.py ..              [ 14%]
beartype_test/a00_unit/a20_util/func/test_utilfuncmake.py ..             [ 14%]
beartype_test/a00_unit/a20_util/func/test_utilfuncscope.py ...           [ 15%]
beartype_test/a00_unit/a20_util/func/test_utilfunctest.py .............  [ 20%]
beartype_test/a00_unit/a20_util/func/test_utilfuncwrap.py .              [ 21%]
beartype_test/a00_unit/a20_util/func/arg/test_utilfuncargget.py .        [ 21%]
beartype_test/a00_unit/a20_util/func/arg/test_utilfuncargiter.py .       [ 22%]
beartype_test/a00_unit/a20_util/func/arg/test_utilfuncargtest.py ...     [ 23%]
beartype_test/a00_unit/a20_util/func/lib/test_utilbeartypefunc.py ..     [ 23%]
beartype_test/a00_unit/a20_util/func/pep/test_utilpep484func.py .        [ 24%]
beartype_test/a00_unit/a20_util/hint/a00_pep/test_a00_utilpepget.py .... [ 25%]
.                                                                        [ 26%]
beartype_test/a00_unit/a20_util/hint/a00_pep/test_a90_utilpeptest.py ... [ 27%]
....                                                                     [ 28%]
beartype_test/a00_unit/a20_util/hint/a00_pep/proposal/test_utilpep544.py . [ 29%]
                                                                         [ 29%]
beartype_test/a00_unit/a20_util/hint/a00_pep/proposal/test_utilpep557.py . [ 29%]
                                                                         [ 29%]
beartype_test/a00_unit/a20_util/hint/a00_pep/proposal/test_utilpep586.py . [ 30%]
                                                                         [ 30%]
beartype_test/a00_unit/a20_util/hint/a00_pep/proposal/test_utilpep593.py . [ 30%]
                                                                         [ 30%]
beartype_test/a00_unit/a20_util/hint/a00_pep/proposal/test_utilpep604.py . [ 30%]
                                                                         [ 30%]
beartype_test/a00_unit/a20_util/hint/a00_pep/proposal/pep484/test_utilpep484typevar.py . [ 31%]
                                                                         [ 31%]
beartype_test/a00_unit/a20_util/hint/a00_pep/proposal/pep484585/test_utilpep484585.py . [ 31%]
                                                                         [ 31%]
beartype_test/a00_unit/a20_util/hint/a00_pep/proposal/pep484585/test_utilpep484585callable.py . [ 31%]
                                                                         [ 31%]
beartype_test/a00_unit/a20_util/hint/a00_pep/proposal/pep484585/test_utilpep484585generic.py . [ 32%]
..                                                                       [ 33%]
beartype_test/a00_unit/a20_util/hint/a00_pep/proposal/pep484585/test_utilpep484585subclass.py . [ 33%]
                                                                         [ 33%]
beartype_test/a00_unit/a20_util/hint/a10_nonpep/test_utilhintnonpeptest.py . [ 33%]
..                                                                       [ 34%]
beartype_test/a00_unit/a20_util/hint/a90_core/test_utilhintfactory.py .  [ 34%]
beartype_test/a00_unit/a20_util/hint/a90_core/test_utilhinttest.py ...   [ 36%]
beartype_test/a00_unit/a20_util/kind/test_utilkinddict.py ......         [ 38%]
beartype_test/a00_unit/a20_util/mod/test_utilmoddeprecate.py .           [ 38%]
beartype_test/a00_unit/a20_util/mod/test_utilmodget.py .                 [ 39%]
beartype_test/a00_unit/a20_util/mod/test_utilmodimport.py ...            [ 40%]
beartype_test/a00_unit/a20_util/mod/test_utilmodtest.py ..               [ 41%]
beartype_test/a00_unit/a20_util/mod/lib/test_utiltyping.py .             [ 41%]
beartype_test/a00_unit/a20_util/os/test_utilostest.py ..                 [ 42%]
beartype_test/a00_unit/a20_util/os/test_utilostty.py .                   [ 42%]
beartype_test/a00_unit/a20_util/py/test_utilpyinterpreter.py .           [ 42%]
beartype_test/a00_unit/a20_util/py/test_utilpyweakref.py ..              [ 43%]
beartype_test/a00_unit/a20_util/py/test_utilpyword.py .                  [ 44%]
beartype_test/a00_unit/a20_util/text/test_utiltextident.py .             [ 44%]
beartype_test/a00_unit/a20_util/text/test_utiltextjoin.py ..             [ 45%]
beartype_test/a00_unit/a20_util/text/test_utiltextlabel.py ..            [ 46%]
beartype_test/a00_unit/a20_util/text/test_utiltextmunge.py ....          [ 47%]
beartype_test/a00_unit/a20_util/text/test_utiltextrepr.py .              [ 47%]
beartype_test/a00_unit/a60_api/test_api_cave.py .....                    [ 49%]
beartype_test/a00_unit/a60_api/test_api_meta.py .                        [ 50%]
beartype_test/a00_unit/a60_api/conf/test_conf.py ..                      [ 50%]
beartype_test/a00_unit/a60_api/door/test_door_func.py .                  [ 51%]
beartype_test/a00_unit/a60_api/door/test_door_typehint.py .............  [ 56%]
beartype_test/a00_unit/a60_api/peps/test_pep563.py .......               [ 58%]
beartype_test/a00_unit/a60_api/plug/test_plugmixin.py .                  [ 59%]
beartype_test/a00_unit/a60_api/typing/test_typingpep544.py .......s      [ 62%]
beartype_test/a00_unit/a60_api/vale/_core/test_valecore.py ..            [ 63%]
beartype_test/a00_unit/a60_api/vale/_is/test_valeis.py ...               [ 64%]
beartype_test/a00_unit/a60_api/vale/_is/test_valeiscls.py ....           [ 65%]
beartype_test/a00_unit/a60_api/vale/_is/test_valeisobj.py ..             [ 66%]
beartype_test/a00_unit/a60_api/vale/_is/test_valeisoper.py ..            [ 67%]
beartype_test/a00_unit/a60_api/vale/_util/test_valeutiltext.py .         [ 67%]
beartype_test/a00_unit/a80_check/a00_expr/test_exprcode.py .             [ 68%]
beartype_test/a00_unit/a80_check/a00_expr/test_exprscope.py .....        [ 69%]
beartype_test/a00_unit/a80_check/a20_conv/test_convcoerce.py ..          [ 70%]
beartype_test/a00_unit/a80_check/a20_conv/test_convreduce.py .           [ 71%]
beartype_test/a00_unit/a80_check/a90_door/test_checkdoor.py ....         [ 72%]
beartype_test/a00_unit/a90_decor/test_decorcall.py .                     [ 73%]
beartype_test/a00_unit/a90_decor/test_decorconf.py ...                   [ 74%]
beartype_test/a00_unit/a90_decor/test_decorwrappee.py ....               [ 75%]
beartype_test/a00_unit/a90_decor/test_decorwrapper.py .                  [ 76%]
beartype_test/a00_unit/a90_decor/a00_cache/test_cachetype.py ..          [ 76%]
beartype_test/a00_unit/a90_decor/a20_error/a00_util/test_errorutiltext.py . [ 77%]
                                                                         [ 77%]
beartype_test/a00_unit/a90_decor/a20_error/a90_main/test_errormain.py .  [ 77%]
beartype_test/a00_unit/a90_decor/a40_code/test_codeargkind.py ......     [ 79%]
beartype_test/a00_unit/a90_decor/a40_code/test_codenoop.py ......        [ 82%]
beartype_test/a00_unit/a90_decor/a40_code/a00_nonpep/test_codemypy.py .  [ 82%]
beartype_test/a00_unit/a90_decor/a40_code/a00_nonpep/test_codenonpep.py . [ 82%]
........                                                                 [ 85%]
beartype_test/a00_unit/a90_decor/a40_code/a90_pep/test_pep484585.py ...  [ 87%]
beartype_test/a00_unit/a90_decor/a40_code/a90_pep/test_pep544.py ......  [ 89%]
beartype_test/a00_unit/a90_decor/a40_code/a90_pep/test_pep557.py .       [ 89%]
beartype_test/a00_unit/a90_decor/a40_code/a90_pep/test_pep561.py .       [ 90%]
beartype_test/a00_unit/a90_decor/a40_code/a90_pep/test_pep585.py ...     [ 91%]
beartype_test/a00_unit/a90_decor/a40_code/a90_pep/test_pep589.py .       [ 91%]
beartype_test/a00_unit/a90_decor/a40_code/a90_pep/test_pep593.py ...     [ 92%]
beartype_test/a00_unit/a90_decor/a40_code/a90_pep/pep484/test_pep484.py . [ 93%]
....                                                                     [ 94%]
beartype_test/a00_unit/a90_decor/a40_code/a90_pep/pep484/test_pep484ref.py . [ 95%]
....                                                                     [ 96%]
beartype_test/a00_unit/a90_decor/a60_main/test_beartype.py .             [ 96%]
beartype_test/a00_unit/a90_decor/a90_roar/test_beartype_roar.py .        [ 97%]
beartype_test/a90_func/doc/test_docreadme.py .                           [ 97%]
beartype_test/a90_func/package/test_package_import.py .                  [ 98%]
beartype_test/a90_func/pep/test_pep561_static.py .s                      [ 98%]
beartype_test/a90_func/z90_lib/a00_sphinx/test_beartype_in_sphinx.py F   [ 99%]
beartype_test/a90_func/z90_lib/a00_sphinx/test_sphinx_build.py s         [ 99%]
beartype_test/a90_func/z90_lib/a90_nuitka/test_nuitka.py s               [100%]

=================================== FAILURES ===================================
___________________________ test_beartype_in_sphinx ____________________________

tmp_path = PosixPath('/tmp/pytest-of-tkloczko/pytest-1/test_beartype_in_sphinx0')

    @skip_if_ci()
    @skip_unless_package('sphinx')
    @ignore_warnings(DeprecationWarning)
    def test_beartype_in_sphinx(tmp_path) -> None:
        '''
        Functional test validating that the :func:`beartype.beartype` decorator
        conditionally reduces to a noop when the active Python interpreter is
        building documentation for the third-party :mod:`sphinx` package.

        To do so, this test externally runs the ``sphinx-build`` command against a
        minimal-length Sphinx document tree exercising all known edge cases.

        Parameters
        ----------
        tmp_path : pathlib.Path
            Abstract path encapsulating a temporary directory unique to this unit
            test, created in the base temporary directory.
        '''

        # ..................{ IMPORTS                            }..................
        # Defer test-specific imports.
        from beartype import beartype
        from beartype._util.mod.lib.utilsphinx import (
            _SPHINX_AUTODOC_SUBPACKAGE_NAME)
        from beartype_test._util.cmd.pytcmdexit import is_success
        from beartype_test._util.path.pytpathtest import (
            get_test_func_data_lib_sphinx_dir)
        from sys import modules as modules_imported_name

        # Entry-point (i.e., pure-Python function accepting a list of zero or more
        # command-line arguments) underlying the external "sphinx-build" command.
        from sphinx.cmd.build import main as sphinx_build

        # ..................{ SPHINX-BUILD                       }..................
        # List of all command-line options (i.e., "-"-prefixed strings) to be
        # effectively passed to the external "sphinx-build" command.
        #
        # Note this iterable *MUST* be defined as a list rather than tuple. If a
        # tuple, the function called below raises an exception. Hot garbage!
        SPHINX_OPTIONS = [
            # Avoid caching data into a "{OUTPUTDIR}/.doctrees/" subdirectory.
            # Although typically advisable, "{OUTPUTDIR}" is an ignorable temporary
            # test-specific directory deleted immediately after completion of this
            # test. Caching data would only needlessly consume time and space.
            '-E',

            # Enable the HTML mode, rendering HTML-specific documentation files.
            # Although technically arbitrary, this is the typical default mode.
            '-b', 'html',

            #FIXME: Reinstate this at some point, please. Sadly, Sphinx is currently
            #emitting ignorable "DeprecationWarning" warnings that we *DO* genuinely
            #need to ignore. Does Sphinx provide some means of treating only *SOME*
            #warnings as errors or is it truly just an unconditional global thing?
            # # Treat non-fatal warnings as fatal errors. This is *CRITICAL.* By
            # # default, Sphinx insanely emits non-fatal warnings for fatal "autodoc"
            # # errors resembling:
            # #     WARNING: autodoc: failed to import module 'beartype_sphinx'; the following exception was raised:
            # #     No module named 'beartype_sphinx'
            # '-W',
        ]

        # List of all command-line arguments (i.e., non-options) to be effectively
        # passed to the external "sphinx-build" command.
        #
        # Note this iterable *MUST* be defined as a list rather than tuple. If a
        # tuple, the function called below raises an exception. Steaming trash!
        SPHINX_ARGUMENTS = [
            # Absolute or relative dirname of a test-specific subdirectory
            # containing a sample Sphinx structure exercising edge cases in the
            # @beartype decorator.
            str(get_test_func_data_lib_sphinx_dir()),
            # Absolute or relative dirname of a test-specific temporary directory to
            # which Sphinx will emit ignorable rendered documentation files.
            str(tmp_path),
        ]

        # Run "sphinx-build" to build documentation for this fake project.
        sphinx_build_exit_code = sphinx_build(SPHINX_OPTIONS + SPHINX_ARGUMENTS)

        # Assert that "sphinx-build" successfully builds documentation for this
        # fake project *WITHOUT* raising an exception.
>       assert is_success(sphinx_build_exit_code), (
            f'"sphinx-build" exit code {sphinx_build_exit_code} != 0.')
E       AssertionError: "sphinx-build" exit code 2 != 0.
E       assert False
E        +  where False = <function is_success at 0x7f980fa44e50>(2)

SPHINX_ARGUMENTS = ['/home/tkloczko/rpmbuild/BUILD/beartype-0.12.0/beartype_test/a90_func/data/lib/sphinx', '/tmp/pytest-of-tkloczko/pytest-1/test_beartype_in_sphinx0']
SPHINX_OPTIONS = ['-E', '-b', 'html']
_SPHINX_AUTODOC_SUBPACKAGE_NAME = 'sphinx.ext.autodoc'
beartype   = <function beartype at 0x7f9996387e50>
get_test_func_data_lib_sphinx_dir = <function get_test_func_data_lib_sphinx_dir at 0x7f980f7fe670>
is_success = <function is_success at 0x7f980fa44e50>
modules_imported_name = {'__future__': <module '__future__' from '/usr/lib64/python3.8/__future__.py'>, '__main__': <module '__main__' from '/usr/bin/pytest'>, '__mp_main__': <module '__main__' from '/usr/bin/pytest'>, '_abc': <module '_abc' (built-in)>, ...}
sphinx_build = <function main at 0x7f980fa44550>
sphinx_build_exit_code = 2
tmp_path   = PosixPath('/tmp/pytest-of-tkloczko/pytest-1/test_beartype_in_sphinx0')

beartype_test/a90_func/z90_lib/a00_sphinx/test_beartype_in_sphinx.py:120: AssertionError
----------------------------- Captured stdout call -----------------------------
Running Sphinx v6.1.3
----------------------------- Captured stderr call -----------------------------

Exception occurred:
  File "/usr/lib/python3.8/site-packages/sphinx/deprecation.py", line 46, in __getattr__
    warnings.warn(f"The alias '{self._modname}.{name}' is deprecated, "
sphinx.deprecation.RemovedInSphinx80Warning: The alias 'sphinx.util.progress_message' is deprecated, use 'sphinx.util.display.progress_message' instead. Check CHANGES for Sphinx API modifications.
The full traceback has been saved in /tmp/sphinx-err-m6mcd1ye.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
=========================== short test summary info ============================
SKIPPED [1] beartype_test/a00_unit/a60_api/typing/test_typingpep544.py:439: Python 3.8.16 < 3.9.0.
SKIPPED [1] beartype_test/a90_func/pep/test_pep561_static.py:222: Command "pyright" not found.
SKIPPED [1] beartype_test/a90_func/z90_lib/a00_sphinx/test_sphinx_build.py:34: Currently broken by Sphinx "autoapi" deprecations.
SKIPPED [1] beartype_test/a90_func/z90_lib/a90_nuitka/test_nuitka.py:31: could not import 'nuitka.__init__': No module named 'nuitka'
FAILED beartype_test/a90_func/z90_lib/a00_sphinx/test_beartype_in_sphinx.py::test_beartype_in_sphinx
================== 1 failed, 258 passed, 4 skipped in 27.00s ===================
```
</details>

Here is list of installed modules in build env
<details>

```console
Package                       Version
----------------------------- -----------------
alabaster                     0.7.13
astroid                       2.14.2
attrs                         22.2.0
Babel                         2.12.1
build                         0.9.0
charset-normalizer            3.1.0
distro                        1.8.0
docutils                      0.19
exceptiongroup                1.0.0
gpg                           1.18.0-unknown
idna                          3.4
imagesize                     1.4.1
importlib-metadata            6.0.0
iniconfig                     2.0.0
Jinja2                        3.1.2
lazy-object-proxy             1.9.0
libcomps                      0.1.19
MarkupSafe                    2.1.2
mypy                          1.0.1
mypy-extensions               0.4.3
numpy                         1.24.2
packaging                     23.0
pep517                        0.13.0
pip                           22.3.1
pluggy                        1.0.0
Pygments                      2.14.0
pytest                        7.2.2
python-dateutil               2.8.2
pytz                          2022.4
PyYAML                        6.0
requests                      2.28.2
rpm                           4.17.0
setuptools                    65.6.3
six                           1.16.0
snowballstemmer               2.2.0
Sphinx                        6.1.3
sphinx-autoapi                1.9.0
sphinxcontrib-applehelp       1.0.4
sphinxcontrib-devhelp         1.0.2.dev20230202
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1.dev20230128
sphinxcontrib-qthelp          1.0.3.dev20230128
sphinxcontrib-serializinghtml 1.1.5
tomli                         2.0.1
typing_extensions             4.4.0
Unidecode                     1.3.4
urllib3                       1.26.12
wheel                         0.38.4
wrapt                         1.14.1
zipp                          3.15.0
```
</details>

**Comment by leycec** [comment]
Thanks so much for packaging @beartype, @kloczek. As a fellow packager, I feel your pain and apologize for what @beartype has done to you. :pray: 

As expected, this is all Sphinx's fault. Since Sphinx deprecates functionality so frequently (*...not necessarily a bad thing in and of itself, of course*), @beartype's Sphinx integration tests tend to frequently fail. This is one of those times.

Thankfully... **we've already resolved this!** Somewhere around commit 50f50d4e5e, I think? *Annnnyway,* we should clearly publish a new `beartype 0.12.1` patch release resolving this and a few other critical test- and installation-time issues.

Let's see whether @leycec can deliver on his lofty and unsubstantiated promises tonight.

**Comment by kloczek** [comment]
Thos commit cannot be applied on top of last release.
Do you have any plans to make the new release soon? 🤔

**Comment by kloczek** [comment]
OK so do you have any plans to make such release soon? 🤔 (ETA? .. just asking)

**Comment by leycec** [comment]
**Tomorrow.** Seriously. It's happening, everybody! Until then, consider omitting tests that are known to fail. As detailed above, the one-liner that brings joy is:

```bash
pytest -k 'not test_beartype_in_sphinx and not test_wrapper_fail_obj_large'
```

Even after `beartype 0.13.0` lands and fixes the broken `test_beartype_in_sphinx` test, you'll still want to ignore the non-broken `test_wrapper_fail_obj_large` test: e.g.,

```bash
pytest -k 'not test_wrapper_fail_obj_large'
```

Why? Because that's a performance regression test. It depends on machine-specific timings that will fail under heavy CPU load. It's fragile. Nobody except me (and GitHub Actions) should be running that test. Right?

## tl;dr

**Friday.** I swear it, @kloczek! :sweat_smile:
